cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/DcLineSegmentConversion.java

```diff
@@ -30,30 +34,78 @@ public DcLineSegmentConversion(PropertyBag l, Context context) {
+        if (decodeMode().equals(HvdcLine.ConvertersMode.SIDE_1_INVERTER_SIDE_2_RECTIFIER) && pAC != 0) {
+            lossFactor = (float) (poleLossP2 / Math.abs(pAC)) * 100;
+        } else if (decodeMode().equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER) && Math.abs(pAC) + poleLossP2 != 0) {
+            lossFactor = (float) (poleLossP2 / (Math.abs(pAC) + poleLossP2)) * 100;
+        }
+        iconverter2.setLossFactor(lossFactor);
+
+        if (iconverter1.getHvdcType() == HvdcConverterStation.HvdcType.LCC) {
+            ((LccConverterStation) iconverter1).setPowerFactor(pAC == 0 || Double.isNaN(iconverter1.getTerminal().getQ()) ? 0.8f : (float) Math.abs(iconverter1.getTerminal().getQ() / pAC));
+            ((LccConverterStation) iconverter2).setPowerFactor(pAC == 0 || Double.isNaN(iconverter2.getTerminal().getQ()) ? 0.8f : (float) Math.abs(iconverter2.getTerminal().getQ() / pAC));
+        }
+    }
+
     @Override
     public boolean valid() {
-        if (iconverter1 == null) {
-            missing("Converter1");
-        }
-        if (iconverter2 == null) {
-            missing("Converter2");
+        if (iconverter1 == null || iconverter2 == null) {
+            if (iconverter1 == null) {
+                missing("Converter1");
+            } else {
+                iconverter1.remove();
+            }
+            if (iconverter2 == null) {
+                missing("Converter2");
+            } else {
+                iconverter2.remove();
+            }
+        } else if (!iconverter1.getHvdcType().equals(iconverter2.getHvdcType())) {
+            iconverter1.remove();
+            iconverter2.remove();
+            invalid(String.format("HVDC converter station %s and HVDC converter station %s are linked and of different types",
+                    iconverter1.getId(), iconverter2.getId()));
+        } else {
+            return true;
         }
-        return iconverter1 != null && iconverter2 != null;
+        return false;
     }
 
     @Override
     public void convert() {
         Objects.requireNonNull(iconverter1);
         Objects.requireNonNull(iconverter2);
 
-        double p = activePowerSetpoint();
-        double maxP = p * 1.2;
+        double pAC1 = Double.isNaN(cconverter1.asDouble(TARGET_PPCC)) ? 0 : cconverter1.asDouble(TARGET_PPCC);
+        double pAC2 = Double.isNaN(cconverter2.asDouble(TARGET_PPCC)) ? 0 : cconverter2.asDouble(TARGET_PPCC);
+        double pAC = Math.abs(decodeMode().equals(HvdcLine.ConvertersMode.SIDE_1_RECTIFIER_SIDE_2_INVERTER) && pAC1 != 0 || pAC2 == 0 ? pAC1 : pAC2);
+        updateConverterStations(pAC);
+
+        double maxP = pAC * 1.2; // arbitrary value because there is no maxP attribute in CGMES
         missing("maxP", maxP);
 
         HvdcLineAdder adder = context.network().newHvdcLine()
                 .setR(r())
                 .setNominalV(ratedUdc())
-                .setActivePowerSetpoint(p)
+                .setActivePowerSetpoint(pAC)
```

Comment:
I think we have to compute P(DC) here, from pAC1/poleLossP1 or pAC2/ poleLossP2, depending on the side of the AC target value.